client/ayon_harmony/plugins/create/create_render_layers.py

```diff
@@ -0,0 +1,1014 @@
+        }
+        pre_create_data: dict[str, Any] = {
+            "group_id": group_id,
+            "mark_for_review": mark_for_review,
+            "render_target": render_target,
+        }
+        return creator.create(product_name, instance_data, pre_create_data)
+    
+    def _prepare_render_passes(
+        self,
+        project_entity: dict[str, Any],
+        folder_entity: dict[str, Any],
+        task_entity: dict[str, Any],
+        render_layer_instance: CreatedInstance,
+        layers: list[dict[str, Any]],
+        mark_for_review: bool,
+        render_target: str,
+        existing_render_passes: list[CreatedInstance],
+    ):
+        task_name = task_entity["name"]
+        creator: CreateRenderPass = self.create_context.creators[
+            CreateRenderPass.identifier
+        ]
+        render_pass_by_layer_name = {}
+        for render_pass in existing_render_passes:
+            render_pass_by_layer_name[render_pass["layer_name"]] = render_pass
+
+        # Use renaming template to parse correct variant from existing layer
+        #   names.
+        name_regex = None
+        if self.layer_name_template["enabled"]:
+            template = self.layer_name_template["template"]
+            fake_group = "___group___"
+            fake_layer = "___layer___"
+            fake_variant = "___variant___"
+            try:
+                name_regex = template.format(
+                    layer_index=fake_layer,
+                    group_index=fake_group,
+                    variant=fake_variant,
+                )
+            except Exception:
+                self.log.error(
+                    "Failed to fill name regex template.",
+                    exc_info=True
+                )
+                name_regex = ""
+
+            for src, regex in (
+                (fake_group, r"(?P<group>\d+)"),
+                (fake_layer, r"(?P<layer>\d+)"),
+                (fake_variant, r"(?P<variant>.*)"),
+            ):
+                name_regex = name_regex.replace(src, regex)
+            name_regex = re.compile(name_regex)
+        layer_positions_in_groups = self._get_layer_positions_in_groups(layers)
+        groups_info = get_group_infos()
+
+        for layer in layers:
+            layer_name = layer["name"]
+            variant = None
+            render_pass = render_pass_by_layer_name.get(layer_name)
+            if render_pass is not None and len(render_pass["layer_names"]) > 0:
+                variant = render_pass["variant"]
+            elif name_regex is not None:
+                result = name_regex.match(layer_name)
+                groups = {}
+                if result is not None:
+                    groups = result.groupdict()
+                variant = groups.get("variant")
+
+            if not variant:
+                variant = layer["name"]
+
+            group_id = get_group_id(layer["color"], groups_info)
+            renderlayer = get_group_layer_name(
```

Comment:
You really should just store render layer instance id to render pass instance and calculate the product name on demand.